### PR TITLE
Refactoring `internal/db/kvs/redis` package

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I refactored `internal/db/kvs/redis` package. Because I can not add test.
In this PR contains changes of core logic and test.
I fixed the logic with author, so I think the merge is fine without confirmation of author.

### Related Card:

https://github.com/vdaas/vald/projects/3#card-42173491

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kevindiu** [comment]
LGTM

File: internal/db/kvs/redis/redis.go [hunk 1]
```diff
@@ -75,89 +83,99 @@ type redisClient struct {
...
+	for {
+		select {
+		case <-pctx.Done():
+			return nil, errors.Wrap(errors.Wrap(err, errors.ErrRedisConnectionPingFailed.Error()), pctx.Err().Error())
+		case <-tick.C:
+			err = rc.client.Ping().Err()
+			if err == nil {
+				return rc.client, nil
 			}
+			log.Error(err)
```
Comment (vankichi) [inline internal/db/kvs/redis/redis.go:178]:
```suggestion
			log.Warn(err)
```

File: internal/db/kvs/redis/redis.go [hunk 2]
```diff
@@ -75,89 +83,99 @@ type redisClient struct {
...
+}
+
+func (rc *redisClient) ping(ctx context.Context) (r Redis, err error) {
+	pctx, cancel := context.WithTimeout(ctx, rc.initialPingTimeLimit)
+	defer cancel()
+	tick := time.NewTicker(rc.initialPingDuration)
+	for {
+		select {
+		case <-pctx.Done():
+			return nil, errors.Wrap(errors.Wrap(err, errors.ErrRedisConnectionPingFailed.Error()), pctx.Err().Error())
```
Comment (vankichi) [inline internal/db/kvs/redis/redis.go:172]:
```suggestion
                         // output error before return
			return nil, errors.Wrap(errors.Wrap(err, errors.ErrRedisConnectionPingFailed.Error()), pctx.Err().Error())
```

**Comment by vankichi** [comment]
@hlts2 when you fix, I'll merge it.

**Pull request merged**